C/plugins/storage/postgres/connection.cpp

```diff
@@ -3312,3 +3314,712 @@ const char *p;
+								else
+								{
+									// alter table case, table already exists
+									// check if column already exists in database
+									if (dbCol.find(v.column) == dbCol.end())
+									{
+										// if it is not a key then add the column else log error
+										if (!v.key)
+										{
+											sql += "add column ";
+											sql += v.column + " " + v.type;
+			                                                                if (v.type == "varchar")
+               				                                                {
+                        	                                                        	sql += "(" + std::to_string(v.sz) + ")";
+											}
+											sql +=",";
+											columnsToAlter = true;
+										}
+										else
+										{
+											// altering a key is not allowed
+											Logger::getLogger()->error("%s:%d altering key request(%s) is not allowed for an existing table", __FUNCTION__, __LINE__, v.column.c_str());
+										}
+									}
+									else
+									{
+										// altering an existing column not alllowed
+                                                                        	Logger::getLogger()->error("%s:%d altering an existing column %s is not allowed", __FUNCTION__, __LINE__, v.column.c_str());
+									}
+								}
+							}
+
+							// If altering the table, drop all the columns which are present
+							// in DB but are not in the request, iterate over DB colums list 
+							// to find out columns which are ppresent in DB, compare with the
+							// incoming request list of columns colsPerTableInReq
+
+							if(alterTable)
+							{
+								for ( auto col : dbCol)
+								{
+									// Make sure the column to be dropped is not a primary key
+									// check in the tbKeyMapFromDB for the key of this table
+									columnRec c;
+									c.column = col;
+									if(colsPerTableInReq.find(c) == colsPerTableInReq.end() && (tbKeyMapFromDB[name] != col))
+									{
+										// this column is in database but not in latest schema request
+										// need to drop this column
+										sql += "drop column " + col +"," ;
+										columnsToAlter = true;
+									}
+								}
+							}
+							
+							// remove last comma
+							if ( sql[sql.size() - 1] == ',')
+							{
+								sql.erase(sql.size()-1);
+							}
+
+							if (alterTable)
+								sql += " ;";
+							else
+								sql += " );";
+
+							// execute the sql here 
+							// if alterTable is true and no columns to Alter , then dont fire the sqlquery
+							// 
+							if (!( alterTable && !columnsToAlter))
+							{
+								rowsAffectedLastCommand = purgeOperation(sql.c_str(), logSection, "CreatingSchema - phase 1, creating/altering tables", false);
+								if (rowsAffectedLastCommand == -1)
+                                                		{
+                                                        		Logger::getLogger()->error("%s:%d Error in creating/altering tables",__FUNCTION__,__LINE__);
```

Comment:
Can you give more information that would be helpful here

Reply:
Fixed